libs/providers/in-memory/src/lib/in-memory-provider.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  EvaluationContext,
+  Provider,
+  JsonValue,
+  ResolutionDetails,
+  FlagNotFoundError,
+  TypeMismatchError,
+  StandardResolutionReasons,
+  GeneralError,
+  FlagValue,
+} from '@openfeature/js-sdk';
+import { FlagConfiguration } from './flag-configuration';
+
+export class InMemoryProvider implements Provider {
+  readonly metadata = {
+    name: 'In-Memory Provider',
+  } as const;
+  private _flagConfiguration: FlagConfiguration;
+
+  constructor(flagConfiguration: FlagConfiguration = {}) {
+    this._flagConfiguration = flagConfiguration;
```

Comment:
It really depends on how you want this to work, but JS is pass-by-ref for objects, meaning that since you're not cloning this in any way, the flag definitions can certainly be changed at runtime by changing the value of the object pass here. This might be surprising given what's in the readme.

Here's a branch off this one demoing what I mean: https://github.com/moredip/js-sdk-contrib/pull/1/files

Reply:
yeah, the `replaceConfiguration(...)` method is a defined way for a client to do that type of runtime change (intended for testing purposes). 

You're right that the README is a bit misleading here. You're also right that it would be a bit safer if we were to clone the config, but TBH I'm not sure it's that likely to cause issues. 

I'm happy to leave this as it is for now, and if someone wants to fix it later we can do so!

Reply:
OK fine I couldn't resist fixing it 🤣 I updated the README and added cloning of the configuration/